File: packages/sdk-python/trustchain/signatures.py
from __future__ import annotations

from typing import Any

from .client import TrustChainClient


class SignaturesResource:
    def __init__(self, client: TrustChainClient) -> None:
        self._client = client

    def create(self, payload: dict[str, Any], *, idempotency_key: str | None = None) -> dict[str, Any]:
        return self._client.request("POST", "/signatures", body=payload, idempotency_key=idempotency_key)

    def get(self, signature_id: str) -> dict[str, Any]:
        return self._client.request("GET", f"/signatures/{signature_id}")

File: packages/sdk-typescript/src/signatures.ts
import type { TrustChainClient } from "./client.js";
import type { CreateSignatureInput, Signature } from "./types.js";

export class SignaturesResource {
  constructor(private readonly client: TrustChainClient) {}

  create(input: CreateSignatureInput, opts?: { idempotencyKey?: string }) {
    return this.client.request<{
      signature: Signature;
      generatedPrivateKeyPem?: string;
    }>({
      method: "POST",
      path: "/signatures",
      body: input,
      idempotencyKey: opts?.idempotencyKey,
    });
  }

  get(id: string) {
    return this.client.request<{ signature: Signature }>({
      method: "GET",
      path: `/signatures/${id}`,
    });
  }
}
